controllers/rcj_soccer_referee_supervisor/referee/supervisor.py

```diff
@@ -341,27 +388,31 @@ def _add_initial_position_noise(
         return translation
 
     def reset_robot_position(self, robot):
-        self.getFromDef(robot).setVelocity([0, 0, 0, 0, 0, 0])
+        self.reset_robot_velocity(robot)
 
         translation = ROBOT_INITIAL_TRANSLATION[robot].copy()
         translation = self._add_initial_position_noise(translation)
 
-        tr_field = self.getFromDef(robot).getField('translation')
-        tr_field.setSFVec3f(translation)
+        self.set_robot_position(robot, translation)
+        self.set_robot_rotation(robot, ROBOT_INITIAL_ROTATION[robot])
 
-        rot_field = self.getFromDef(robot).getField('rotation')
-        rot_field.setSFRotation(ROBOT_INITIAL_ROTATION[robot])
+        self.reset_checkers(robot)
 
-        # Ensure the progress checker does not count this "jump"
-        self.progress_chck[robot].reset()
+    def reset_checkers(self, object_name: str):
+        """Reset rule checkers for the specified object.
 
-        self.penalty_area_chck[robot].reset()
+        Args:
+            object_name (str): Either "bal" or the robot's name.
+        """
+        self.progress_chck[object_name].reset()
+        if object_name != 'ball':
+            self.penalty_area_chck[object_name].reset()
 
     def robot_name_to_team_name(self, robot_name: str) -> str:
         if robot_name.startswith('Y'):
-            return self.team_name_yellow
+            return Team.YELLOW.value
```

Comment:
@Adman I don't think this is right. The purpose of this function is to return the actual name of the team so that it could get logged into the reflog and so on -- I suppose `Team.BLUE.value` will end up being the same in all runs.

Reply:
Yeah, sorry, will revert it back.